diff --git a/composer_manager.install b/composer_manager.install
index bf07daa..135ecaa 100644
--- a/composer_manager.install
+++ b/composer_manager.install
@@ -64,15 +64,8 @@ function composer_manager_requirements($phase) {
 }
 
 /**
- * Converts variables to config entries.
- *
- * Implements hook_update_N().
+ * Implements hook_install().
  */
-function composer_manager_update_8000() {
-  update_variables_to_config('composer_manager.settings', array(
-    'composer_manager_autobuild_file' => 'autobuild_file',
-    'composer_manager_file_dir' => 'file_dir',
-    'composer_manager_vendor_dir' => 'vendor_dir',
-    'composer_manager_autobuild_packages' => 'autobuild_packages',
-  ));
+function composer_manager_install() {
+  \Drupal::service('composer_manager.manager')->registerAutoLoader();
 }
